release-tool.ps1

```diff
@@ -0,0 +1,456 @@
+}
+
+function Invoke-SignFiles([string[]] $files, [string] $key, [string] $time) {
+    if (-not (Test-Path -Path "$key" -PathType leaf)) {
+        throw "Appsign key file was not found! ($key)"
+    }
+    if ($files.Length -eq 0) {
+        return
+    }
+
+    Write-Host "Signing files using $key"  -ForegroundColor Cyan
+    $KeyPassword = Read-Host "Key password: " -MaskInput
+
+    foreach ($_ in $files) {
+        Write-Host "Signing file '$_' using Microsoft signtool..."
+        Invoke-Cmd "signtool" "sign -f `"$key`" -p `"$KeyPassword`" -d `"KeePassXC`" -td sha256 -fd sha256 -tr `"$time`" `"$_`"" -maskargs
+    }
+}
+
+function Invoke-GpgSignFiles([string[]] $files, [string] $key) {
+    if ($files.Length -eq 0) {
+        return
+    }
+
+    Write-Host "Signing files using GPG key $key" -ForegroundColor Cyan
+
+    foreach ($_ in $files) {
+        Write-Host "Signing file '$_' and creating DIGEST..."
+        Remove-Item "$_.sig"
+        Invoke-Cmd "gpg" "--output `"$_.sig`" --armor --local-user `"$key`" --detach-sig `"$_`""
+        $FileName = (Get-Item $_).Name
+        (Get-FileHash "$_" SHA256).Hash + " *$FileName" | Out-File "$_.DIGEST" -NoNewline
+    }
+}
+
+
+# Handle errors and restore state
+$OrigDir = $(Get-Location).Path
+$OrigBranch = $(git rev-parse --abbrev-ref HEAD)
+$ErrorActionPreference = 'Stop'
+trap {
+    Write-Host "Restoring state..." -ForegroundColor Yellow
+    $(git checkout $OrigBranch)
+    Set-Location "$OrigDir"
+}
+
+Write-Host "KeePassXC Release Preparation Helper" -ForegroundColor Green
+Write-Host "Copyright (C) 2021 KeePassXC Team <https://keepassxc.org/>`n" -ForegroundColor Green
+
+# Prepend extra PATH locations as specified
+if ($ExtraPath) {
+    $env:Path = "$ExtraPath;$env:Path"
+}
+
+# Resolve absolute directory for paths
+$SourceDir = (Resolve-Path $SourceDir).Path
+
+# Check format of -Version
+if ($Version -notmatch "^\d+\.\d+\.\d+$") {
+    throw "Invalid format for -Version input"
+}
+
+# Check platform
+if (-not $IsWindows) {
+    throw "The powershell release tool is not available for Linux or macOS at this time."
+}
+
+if ($Merge) {
+    Test-RequiredPrograms
+
+    # Change to SourceDir
+    Set-Location "$SourceDir"
+
+    Test-VersionInFiles
+    Test-WorkingTreeClean
+
+    if (-not $SourceBranch.Length) {
+        $SourceBranch = $(git branch --show-current)
+    }
+
+    if ($SourceBranch -notmatch "^release/.*|develop$") {
+        throw "Must be on a Release or Develop branch to continue with merge."
+    }
+
+    # Update translation files
+    Write-Host "Updating source translation file..."
+    Invoke-Cmd "lupdate" "-no-ui-lines -disable-heuristic similartext -locations none", `
+        "-no-obsolete ./src -ts share/translations/keepassxc_en.ts"
+
+    Write-Host "Pulling updated translations from Transifex..."
+    Invoke-Cmd "tx" "pull -af --minimum-perc=60 --parallel -r keepassxc.share-translations-keepassxc-en-ts--develop"
+
```

Comment:
This should be a separate command that can also be called individually by the user.

Reply:
I want to move this into the cmake build (similar to format) because it is incredibly annoying to have to do this outside of the build process prior to pushing a PR.